### PR TITLE
batchInsert pkNotAI support undo

### PR DESCRIPTION
**Pull request by zjinlei**
### Ⅰ. Describe what this PR did
Support for rollback of non-self-incrementing primary key bulk inserts

### CODE REVIEW

File: rm-datasource/src/main/java/io/seata/rm/datasource/exec/InsertExecutor.java [hunk 1]
```diff
@@ -82,30 +83,50 @@ protected TableRecords afterImage(TableRecords beforeImage) throws SQLException
     }
 
     protected boolean containsPK() {
-        SQLInsertRecognizer recogizier = (SQLInsertRecognizer)sqlRecognizer;
+        SQLInsertRecognizer recogizier = (SQLInsertRecognizer) sqlRecognizer;
```
Comment (xingfudeshi) [inline rm-datasource/src/main/java/io/seata/rm/datasource/exec/InsertExecutor.java]: recogizier->recognizer

File: rm-datasource/src/main/java/io/seata/rm/datasource/exec/InsertExecutor.java [hunk 2]
```diff
@@ -82,30 +83,50 @@ protected TableRecords afterImage(TableRecords beforeImage) throws SQLException
...
         List<String> insertColumns = recogizier.getInsertColumns();
         TableMeta tmeta = getTableMeta();
         return tmeta.containsPK(insertColumns);
     }
 
     protected List<Object> getPkValuesByColumn() throws SQLException {
         // insert values including PK
-        SQLInsertRecognizer recogizier = (SQLInsertRecognizer)sqlRecognizer;
+        SQLInsertRecognizer recogizier = (SQLInsertRecognizer) sqlRecognizer;
```
Comment (xingfudeshi) [inline rm-datasource/src/main/java/io/seata/rm/datasource/exec/InsertExecutor.java]: recogizier->recognizer

File: rm-datasource/src/test/java/io/seata/rm/datasource/exec/BatchInsertExecutorTest.java [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import static org.mockito.Mockito.*;
```
Comment (xingfudeshi) [inline rm-datasource/src/test/java/io/seata/rm/datasource/exec/BatchInsertExecutorTest.java]: Please DO NOT import *.

File: rm-datasource/src/main/java/io/seata/rm/datasource/exec/InsertExecutor.java (hunk 1 above)
Comment (zjinlei) [reply rm-datasource/src/main/java/io/seata/rm/datasource/exec/InsertExecutor.java]: done

File: rm-datasource/src/main/java/io/seata/rm/datasource/exec/InsertExecutor.java (hunk 2 above)
Comment (zjinlei) [reply rm-datasource/src/main/java/io/seata/rm/datasource/exec/InsertExecutor.java]: done

File: rm-datasource/src/test/java/io/seata/rm/datasource/exec/BatchInsertExecutorTest.java (hunk 3 above)
Comment (zjinlei) [reply rm-datasource/src/test/java/io/seata/rm/datasource/exec/BatchInsertExecutorTest.java]: done

**Comment by zjinlei** [comment]
embedded other pr, I will use new pr.

**Comment by zjinlei** [comment]
embedded other pr, I will use new pr.

**Pull request closed without merging**